### ISSUE TITLE
DSOAL integration

### ISSUE THREAD

**Issue by Hiradur** [lutris/agora#83]
DSOAL is an open source DirectSound3D to OpenAL wrapper which brings back proper positional audio (Edit: Wine's DS3D implementation might already do this) and EAX effects to old games relying on DS3D. It consists of two dlls and it would be nice if those could be enabled/disabled per game like dxvk.
I don't think there is an official release yet though.

Git repository for the code: https://github.com/kcat/dsoal
Binaries: https://kcat.strangesoft.net/dsoal.zip

**Comment by raven-gh** [comment]
> Do I need a 7.1 surround sound system to QA this?

No, with using an old game, and looking at the logs, you will see if the game has discovered the emulated sound card HW or not, and enabled EAX.

#### Example: IL-2 1946

Grab the latest build from http://vaporeon.io/hosted/dsoal-builds and drop it in the game folder.
Enjoy EAX 3 (see logs). Pretty noticeable in doppler effects.
This creates: alsoft.ini dsoal-latest.7z dsound.dll
See the newly dropped alsoft.ini config for possible config options, but the default work.
On the logs, you should now see:

```
Initializing DirectSound playback device...
Primary buffer created.
Playback format is set : sampling rate = 44100, num channels = 2.
Buffer caps : Transfer rate = 4096, CPU overhead = 0.
Default speaker config is : 1310724.
Direct sound audio device initialized successfully :                  <- and I don't have a Creative sound card!
DX Version : 7
Hardware    - enabled [buffers : 128]
Extensions  - enabled :
  EAX ver. 1 [X]  - enabled
  EAX ver. 2 [X]  - enabled
  EAX ver. 3 [X]  - enabled                                           <- yay!
  I3D ver. 2 [ ]  - disabled
  ZoomFX     [ ]  - disabled
  MacroFX    [ ]  - disabled
SIMD render [X]
num channels 32
```

**Comment by QmwJlHuSg9pa** [comment]
In my experience of simulating 3D audio for headphones with OpenAL Soft, `stereo-mode` does not always behave as expected. So, it may ideal to explicitly set the following values in `alsoft.conf` (Linux native) or `alsoft.ini` (Wine) to ensure proper behaviour (of course, `drivers=-dsound` is only relevant for Wine applications):

```
[general]
drivers=-dsound,
stereo-mode=headphones
channels=stereo
```

Or, if one wanted to output through a speaker system, then try:
```
[general]
drivers=-dsound,
stereo-mode=speakers
channels= # Autodetect, else specify channel output.
```
